subsys/fs/fuse_fs_access.c

```diff
@@ -0,0 +1,530 @@
+}
+
+static int fuse_fs_access_write(const char *path, const char *buf, size_t size,
+		off_t off, struct fuse_file_info *fi)
+{
+	int err;
+
+	ARG_UNUSED(path);
+
+	if (fi->fh == INVALID_FILE_HANDLE) {
+		return -EINVAL;
+	}
+
+	err = fs_seek(&files[fi->fh], off, FS_SEEK_SET);
+	if (err != 0) {
+		return err;
+	}
+
+	err = fs_write(&files[fi->fh], buf, size);
+
+	return err;
+}
+
+static int fuse_fs_access_ftruncate(const char *path, off_t size,
+				struct fuse_file_info *fi)
+{
+	int err;
+
+	ARG_UNUSED(path);
+
+	if (fi->fh == INVALID_FILE_HANDLE) {
+		return -EINVAL;
+	}
+
+	err = fs_truncate(&files[fi->fh], size);
+
+	return err;
+}
+
+static int fuse_fs_access_truncate(const char *path, off_t size)
+{
+	int err;
+	static struct fs_file_t file;
+
+	err = fs_open(&file, path);
+	if (err != 0) {
+		return err;
+	}
+
+	err = fs_truncate(&file, size);
+	if (err != 0) {
+		fs_close(&file);
+		return err;
+	}
+
+	err = fs_close(&file);
+
+	return err;
+}
+
+static int fuse_fs_access_mkdir(const char *path, mode_t mode)
+{
+	ARG_UNUSED(mode);
+
+	return fs_mkdir(path);
+}
+
+static int fuse_fs_access_rmdir(const char *path)
+{
+	return fs_unlink(path);
+}
+
+static int fuse_fs_access_unlink(const char *path)
+{
+	return fs_unlink(path);
+}
+
+static int fuse_fs_access_statfs(const char *path, struct statvfs *buf)
+{
+	ARG_UNUSED(path);
+	ARG_UNUSED(buf);
+	return 0;
+}
+
+static int fuse_fs_access_utimens(const char *path, const struct timespec tv[2])
+{
+	/* dummy */
+	ARG_UNUSED(path);
+	ARG_UNUSED(tv);
+
+	return 0;
+}
+
+
+static struct fuse_operations fuse_fs_access_oper = {
+	.getattr = fuse_fs_access_getattr,
+	.readlink = NULL,
+	.getdir = NULL,
+	.mknod = NULL,
+	.mkdir = fuse_fs_access_mkdir,
+	.unlink = fuse_fs_access_unlink,
+	.rmdir = fuse_fs_access_rmdir,
+	.symlink = NULL,
+	.rename = NULL,
+	.link = NULL,
+	.chmod = NULL,
+	.chown = NULL,
+	.truncate = fuse_fs_access_truncate,
+	.utime = NULL,
+	.open = fuse_fs_access_open,
+	.read = fuse_fs_access_read,
+	.write = fuse_fs_access_write,
+	.statfs = fuse_fs_access_statfs,
+	.flush = NULL,
+	.release = fuse_fs_access_release,
+	.fsync = NULL,
+	.setxattr = NULL,
+	.getxattr = NULL,
+	.listxattr = NULL,
+	.removexattr = NULL,
+	.opendir  = NULL,
+	.readdir = fuse_fs_access_readdir,
+	.releasedir = NULL,
+	.fsyncdir = NULL,
+	.init = NULL,
+	.destroy = NULL,
+	.access = NULL,
+	.create = fuse_fs_access_create,
+	.ftruncate = fuse_fs_access_ftruncate,
+	.fgetattr = NULL,
+	.lock = NULL,
+	.utimens = fuse_fs_access_utimens,
+	.bmap = NULL,
+	.flag_nullpath_ok = 0,
+	.flag_nopath = 0,
+	.flag_utime_omit_ok = 0,
+	.flag_reserved = 0,
+	.ioctl = NULL,
+	.poll = NULL,
+	.write_buf = NULL,
+	.read_buf = NULL,
+	.flock = NULL,
+	.fallocate = NULL,
+};
+
+static void *fuse_fs_access_main(void *arg)
+{
+	ARG_UNUSED(arg);
+
+	char *argv[] = {
+		"",
+		"-f",
+		"-s",
+		(char *) fuse_mountpoint
+	};
+	int argc = ARRAY_SIZE(argv);
+
+	printf("Mounting flash at %s/\n", fuse_mountpoint);
```

Comment:
nit: you may want `posix_print_trace()` to be sure this ends up in the same place as all other prints in the posix arch